extensions/wikia/Search/classes/FandomSearch.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Wikia\Search;
+
+use \Wikia\Logger\WikiaLogger;
+
+class FandomSearch {
+	const FANDOM_API = 'http://fandom.wikia.com/wp-json/wp/v2/';
+	const FANDOM_API_POSTS_ENDPOINT = 'posts';
+	const FANDOM_SEARCH_PAGE = 'http://fandom.wikia.com/?s=';
+	const RESULTS_COUNT = 5;
+	const VERTICALS_PARENT_CATEGORY_ID = 3;
+	const CATEGORY_TAXONOMY_NAME = 'category';
+	const CACHE_TTL = 12 * 60 * 60;
+
+	public static function getStoriesWithCache( $query ) {
+		return \WikiaDataAccess::cache(
+			wfSharedMemcKey( 'FandomSearch', $query ),
+			static::CACHE_TTL,
+			function() use ( $query ) {
+				return static::getStories( $query );
+			}
+		);
+	}
+
+	public static function getStories( $query ) {
+		$url = static::buildUrl( $query );
+		$method = 'GET';
+		$options = [
+			'curlOptions' => [
+				// TODO set when we use internal copy
+				//CURLOPT_TIMEOUT_MS => 1000,
+				CURLOPT_NOSIGNAL => true,
+			],
+			'noProxy' => true
+		];
+		$log = WikiaLogger::instance();
+
+		$result = \Http::request( $method, $url, $options );
+		if ( $result === false ) {
+			$loggingParams['request'] = [
+				'url' => $url,
+				'method' => $method,
+				'options' => $options,
+			];
+			$log->debug( '', $loggingParams );
+		} else if ( !empty( $result ) ) {
+			return static::parseResult( $result );
+		}
```

Comment:
What is the default return value of this method (i.e. if `$result === false` or if it's otherwise empty)?